### PR TITLE
pkp/pkp-lib#7682 Expand DOI filter functionality

### PR DESCRIPTION
**Pull request by ewhanson**

### CODE REVIEW

File: classes/context/Context.inc.php [hunk 1]
```diff
@@ -68,6 +68,14 @@ public function isDoiTypeEnabled(string $doiType): bool
         return in_array($doiType, $this->getData(Context::SETTING_ENABLED_DOI_TYPES));
     }
 
+    /**
+     * Retrieves array of enabled DOI types (items are one of Repo::doi()::TYPE_*).
+     */
+    public function getEnabledDoiTypes(): ?array
```
Comment (NateWr) [inline classes/context/Context.inc.php:74]: One of the benefits of writing a "getter method" is to be able to use type hinting. It's no big deal, but I might be inclined to return an empty array if no doi types are enabled:

```php
return $this->getData(...) ?? [];
```

That way you get to remove the null possibility from the return type, and the "consumer" of the method doesn't have to perform any type checks.

File: classes/submission/Collector.inc.php [hunk 2]
```diff
@@ -55,8 +55,9 @@ abstract class Collector implements CollectorInterface
     public ?array $stageIds = null;
     public ?array $doiStatuses = null;
 
-    /** @var bool Whether null/empty values should count when considering Doi::STATUS_UNREGISTERED */
-    public bool $strictDoiStatusFilter = false;
+    public ?bool $hasDois = null;
+    /** @var array Which DOI types should be considered when checking if a submission has DOIs set */
+    public array $enabledDoiTypes = [];
```
Comment (NateWr) [inline classes/submission/Collector.inc.php:60]: When there's a comment for a property, add an empty line above. (You can remove the empty line above $hasDois unless you want to add a comment.)

File: classes/submission/Collector.inc.php [hunk 3]
```diff
@@ -88,17 +89,29 @@ public function filterByCategoryIds(?array $categoryIds): AppCollector
...
     }
 
+    /**
+     * Limit results to submissions that do/don't have any DOIs assign to their sub objects
+     *
+     * @param bool|null $hasDois
+     * @param array $enabledDoiTypes TYPE_* constants to consider when checking submission has DOIs
+     * @return AppCollector
+     */
+    public function filterByHasDois(?bool $hasDois, array $enabledDoiTypes = [\APP\doi\Repository::TYPE_PUBLICATION]): AppCollector
```
Comment (NateWr) [inline classes/submission/Collector.inc.php:108]: `[\APP\doi\Repository::TYPE_PUBLICATION]` is great. However, repo facade allows anyone to swap out one repo class for another. To be really safe here, we should always go through the facade. 

I'd recommend leaving `$enabledDoiTypes` null if not set, and then setting the default in the method by going through the `Repo::doi()` facade. If someone wants to override the default to perform a "reset", they can pass an empty array.

File: classes/components/listPanels/PKPDoiListPanel.inc.php [hunk 4]
```diff
@@ -71,12 +71,21 @@ public function getConfig()
         $config['registrationAgencyInfo'] = $this->registrationAgencyInfo;
         $config['doiPrefix'] = $this->doiPrefix;
         $config['filters'][] = [
-            'heading' => __('manager.dois.depositStatus'),
             'filters' => [
+                [
+                    'title' => 'Needs DOI',
```
Comment (asmecher) [inline classes/components/listPanels/PKPDoiListPanel.inc.php:76]: This should be localized, no?

Comment (ewhanson) [reply classes/components/listPanels/PKPDoiListPanel.inc.php]: Yes! Thanks for catching it. Fixed now.

**Pull request merged**